packages/db-kit/src/cli/menu/Menu.tsx

```diff
@@ -0,0 +1,147 @@
+import React, { useState, useEffect } from "react";
+import { Box, Text } from "ink";
+import SelectInput from "ink-select-input";
+import type { Transaction, TransactionReceipt } from "web3-core";
+import type TruffleConfig from "@truffle/config";
+import type { Db, Project } from "@truffle/db";
+
+import type { Definitions, ModeInputProps, ModeName } from "./types";
+
+import * as DecodeAddress from "@truffle/db-kit/cli/decodeAddress";
+import * as DecodeTransaction from "@truffle/db-kit/cli/decodeTransaction";
+
+export interface Props {
+  config: TruffleConfig;
+  db: Db;
+  project: Project.Project;
+  onDone: () => void;
+}
+
+export type MenuModes = {
+  "decode-transaction": {
+    producesEffect: false;
+    rendersComponent: true;
+    props: Props & {
+      transactionHash: string;
+      transaction: Transaction;
+      receipt: TransactionReceipt;
+      addresses: string[];
+    };
+    inputPropName: "transactionHash" | "transaction" | "receipt" | "addresses";
+  };
+  "decode-address": {
+    producesEffect: false;
+    rendersComponent: true;
+    props: Props & {
+      address: string;
+    };
+    inputPropName: "address";
+  };
+  quit: {
+    rendersComponent: false;
+    producesEffect: true;
+    props: Props;
+  };
+};
+
+export const definitions: Definitions<MenuModes> = {
+  "decode-transaction": {
+    label: "Decode transaction",
+    propsInputComponent: props => {
+      return <DecodeTransaction.Inputs {...props} />;
+    },
+    screenComponent: props => {
+      return <DecodeTransaction.Splash {...props} />;
+    }
+  },
+  "decode-address": {
+    label: "Decode contract address",
+    propsInputComponent: props => {
+      return <DecodeAddress.Inputs {...props} />;
+    },
+    screenComponent: props => {
+      return <DecodeAddress.Splash {...props} />;
+    }
+  },
+  quit: {
+    label: "Quit",
+    effect: ({ onDone }) => {
+      onDone();
+    }
+  }
+};
+
+export const Menu = (props: Props) => {
+  const { config, db, project, onDone } = props;
+  const [mode, setMode] = useState<"wait" | ModeName<MenuModes>>("wait");
+
+  const handleSelect = ({ value }) => {
+    setMode(value);
+  };
+
+  const [element, setElement] = useState(<></>);
+  const [inputProps, setInputProps] = useState<
+    ModeInputProps<MenuModes, ModeName<MenuModes>> | undefined
+  >(undefined);
+
+  useEffect(() => {
+    setInputProps(undefined);
+  }, [mode]);
```

Comment:
I don't really get why `mode` is in the dep array here, when it's not being used in the effect. The eslint rules will help here.

Reply:
I probably don't understand how that dependency array works. What I intend here is: every time the mode changes, reset the input props to be empty. Is this not correct?

Reply:
Anyway eslint seems fine with this? Let me know if you're still incredulous!

Reply:
If the linter is fine, I'm fine haha